src/spotify_to_tidal/sync.py

```diff
@@ -222,57 +243,79 @@ def get_tracks_for_new_tidal_playlist(spotify_tracks: Sequence[t_spotify.Spotify
             seen_tracks.add(tidal_id)
     return output
 
-async def sync_playlist(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Session, spotify_playlist, tidal_playlist: tidalapi.Playlist | None, config):
+async def sync_tracks(spotify_tracks: Sequence[t_spotify.SpotifyTrack], old_tidal_tracks: Sequence[tidalapi.Track], config, tidal_session: tidalapi.Session, tidal_playlist: tidalapi.UserPlaylist, sync_favorites: bool = False):
```

Comment:
The function signature is quite different from the one I recommended. You should move more code out of the helper function and into the calling functions until it matches the function signature

Reply:
In my last commit, I refactored this function to extract the different responsibilities into separate helper functions. Please let me know if this is fine. Otherwise, I could move the `get_new_spotify_tracks` retrieval and the `search_new_tracks_on_tidal` retrieval out of the function to better match the signature, but then it would make more sense to remove this function entirely.

Reply:
Hi sorry I have quite specific tastes when it comes to organising code, so easier for me to do it myself than try to explain to you. Here's a patch file:

[cleanup.patch](https://github.com/user-attachments/files/15745962/cleanup.patch)

Reply:
(I haven't tested it thoroughly)